scripts/docs_config/introduction.rst

```diff
@@ -0,0 +1,100 @@
+fine-grained memory allocations. UMF defines distinct interfaces for both pool 
+allocators and memory providers. Users can use pool allocators and memory 
+providers provided by UMF or create their own.
+
+.. figure:: ../assets/images/intro_architecture.png
+
+The UMF library contains various pool allocators and memory providers but also 
+allows for the integration of external ones, giving users the flexibility to 
+either use existing solutions or provide their implementations. 
+
+Memory Providers
+================
+
+A memory provider is an abstraction for coarse (memory page) allocations and 
+deallocations of target memory types, such as host CPU, GPU, or CXL memory. 
+A single distinct memory provider can efficiently operate the memory of devices 
+on the platform or other memory sources such as file-backed or user-provider 
+memory.
+
+UMF comes with several bundled memory providers. Please refer to `README.md`_ 
+to see a full list of them. There is also a possibility to use externally 
+defined memory providers if they implement the UMF interface.
+
+To instantiate a memory provider, user must pass an additional context which 
+contains the details about the specific memory target that should be used. This 
+would be a NUMA node mask for the OS memory provider, file path for the 
+file-backed memory provider, etc. After creation, the memory provider context
+can't be changed.
+
+Pool Allocators
+===============
+
+A pool allocator is an abstraction over object-level memory management based 
+on coarse chunks acquired from the memory provider. It manages the memory pool 
+and services fine-grained malloc/free requests. 
+
+Pool allocators can be implemented to be general purpose or to fulfill 
+specific use cases. Implementations of the pool allocator interface can 
+leverage existing allocators (e.g., jemalloc or oneTBB) or be fully 
+customizable. The pool allocator abstraction could contain basic memory 
+management interfaces, as well as more complex ones that can be used, for 
+example, by the implementation for page monitoring or control (e.g., `madvise`).
+
+UMF comes with several bundled pool allocators. Please refer to `README.md`_ 
+to see a full list of them. There is also a possibility to use externally 
+defined pool allocators if they implement the UMF interface.
+
+Memory Pools
+============
+
+A memory pool consists of a pool allocator and a memory provider instancies 
+along with their properties and allocation policies. Memory pools are used by 
+the `allocation API`_ as a first argument. There is also a possibility to 
+retrieve a memory pool from an existing memory pointer that points to a memory 
+previously allocated by UMF.
+
+.. _UMF: https://github.com/oneapi-src/unified-memory-framework
+.. _README.md: https://github.com/oneapi-src/unified-memory-framework/blob/main/README.md
+.. _allocation API: https://bratpiorka.github.io/unified-memory-framework/api.html#memory-pool
```

Comment:
heh, the link is not a proper one, I believe 😉

Reply:
done